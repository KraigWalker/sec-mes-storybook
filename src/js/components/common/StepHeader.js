import React from 'react';
import { PropTypes } from 'react';
import _ from 'lodash';
import SvgIcon from './GetIcon';
import { Link } from 'react-router-dom';
// import AccessabilityUtils from '../components/common/AccessabilityUtils';

class StepHeader extends React.Component {

	getSubtext(){
		let subText = [];
		_.map(this.props.headerSubtext, (item, index) => {
			subText.push(<p className="c-step-header__subtext" dangerouslySetInnerHTML={{ __html: item }} key={index}></p>);
		});
		return subText;
	}
	render() {
		return (
		<div className="c-step-header" id={this.props.id ? this.props.id : ""}>
		{this.props.showheaderCrumbs ?	<p className="c-step-header__crumbs">
			<Link to = {this.props.headerCrumbsPath} className="c-step-header__link">
				<span className="c-step-header__linkicon"><SvgIcon id="icon-left" width="16px" height="16px"/></span>
				<span className="c-step-header__linktext">{this.props.headerCrumbsMessage}</span>
			</Link>
			</p> : null}
			<h1 className="c-step-header__title">{this.props.headerTitle}</h1>
            {this.props.headerSubtitle ? <p className="c-step-header__subtitle">{this.props.headerSubtitle}</p> : null}
			{this.props.headerSubtext ? this.getSubtext() : null}
		</div>
		);
	}
}

StepHeader.defaultProps = {
	showheaderCrumbs:true,
	showControl:false,
};
export default StepHeader;