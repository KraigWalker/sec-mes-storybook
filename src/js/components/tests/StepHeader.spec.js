import React from 'react';
import { shallow, mount } from 'enzyme';
import StepHeader from '../common/StepHeader';
import { Router } from 'react-router-dom';
import createBrowserHistory from 'history/createBrowserHistory';

const history = createBrowserHistory()

describe('StepHeader Component Check', () => {

  const componentWrap = mount(<Router history={history}><StepHeader headerCrumbsPath="/test"
    headerTitle='title' showControl={false} showheaderCrumbs={true} /></Router>);

  it("StepHeader component should mount", () => {
    expect(typeof componentWrap).toBe('object');
  });
  it("Props Check", () => {
    expect(componentWrap.children().props().showheaderCrumbs).toEqual(true);
    expect(componentWrap.children().props().showControl).toEqual(false);
  });
  it("StepHeader verification", () => {
    expect(componentWrap.find('.c-step-header')).toHaveLength(1);
    expect(componentWrap.find('.c-step-header__crumbs')).toHaveLength(1);
    expect(componentWrap.find('.c-step-header__title')).toHaveLength(1);
    expect(componentWrap.find('.c-step-header__subtext')).toHaveLength(1);
    expect(componentWrap.find('.c-step-header__title').text()).toBe('title');
  });
  it('should render correctly', () => {
    const tree = shallow(
      <StepHeader />
    );
    expect(tree).toMatchSnapshot();
  });
});
