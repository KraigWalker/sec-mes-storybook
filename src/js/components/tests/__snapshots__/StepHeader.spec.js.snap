// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepHeader Component Check should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StepHeader
    showControl={false}
    showheaderCrumbs={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="c-step-header__crumbs"
        >
          <Link
            className="c-step-header__link"
            replace={false}
            to={undefined}
          >
            <span
              className="c-step-header__linkicon"
            >
              <CommonGetIcon
                height="16px"
                id="icon-left"
                width="16px"
              />
            </span>
            <span
              className="c-step-header__linktext"
            />
          </Link>
        </p>,
        <h1
          className="c-step-header__title"
          id="headingTag"
          tabIndex="-1"
        />,
        null,
        <p
          className="c-step-header__subtext"
        />,
      ],
      "className": "c-step-header",
      "id": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="c-step-header__link"
            replace={false}
            to={undefined}
          >
            <span
              className="c-step-header__linkicon"
            >
              <CommonGetIcon
                height="16px"
                id="icon-left"
                width="16px"
              />
            </span>
            <span
              className="c-step-header__linktext"
            />
          </Link>,
          "className": "c-step-header__crumbs",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span
                className="c-step-header__linkicon"
              >
                <CommonGetIcon
                  height="16px"
                  id="icon-left"
                  width="16px"
                />
              </span>,
              <span
                className="c-step-header__linktext"
              />,
            ],
            "className": "c-step-header__link",
            "replace": false,
            "to": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <CommonGetIcon
                  height="16px"
                  id="icon-left"
                  width="16px"
                />,
                "className": "c-step-header__linkicon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "height": "16px",
                  "id": "icon-left",
                  "width": "16px",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "c-step-header__linktext",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "c-step-header__title",
          "id": "headingTag",
          "tabIndex": "-1",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "c-step-header__subtext",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="c-step-header__crumbs"
          >
            <Link
              className="c-step-header__link"
              replace={false}
              to={undefined}
            >
              <span
                className="c-step-header__linkicon"
              >
                <CommonGetIcon
                  height="16px"
                  id="icon-left"
                  width="16px"
                />
              </span>
              <span
                className="c-step-header__linktext"
              />
            </Link>
          </p>,
          <h1
            className="c-step-header__title"
            id="headingTag"
            tabIndex="-1"
          />,
          null,
          <p
            className="c-step-header__subtext"
          />,
        ],
        "className": "c-step-header",
        "id": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="c-step-header__link"
              replace={false}
              to={undefined}
            >
              <span
                className="c-step-header__linkicon"
              >
                <CommonGetIcon
                  height="16px"
                  id="icon-left"
                  width="16px"
                />
              </span>
              <span
                className="c-step-header__linktext"
              />
            </Link>,
            "className": "c-step-header__crumbs",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <span
                  className="c-step-header__linkicon"
                >
                  <CommonGetIcon
                    height="16px"
                    id="icon-left"
                    width="16px"
                  />
                </span>,
                <span
                  className="c-step-header__linktext"
                />,
              ],
              "className": "c-step-header__link",
              "replace": false,
              "to": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <CommonGetIcon
                    height="16px"
                    id="icon-left"
                    width="16px"
                  />,
                  "className": "c-step-header__linkicon",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "height": "16px",
                    "id": "icon-left",
                    "width": "16px",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "c-step-header__linktext",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "c-step-header__title",
            "id": "headingTag",
            "tabIndex": "-1",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "c-step-header__subtext",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
